### PR TITLE
Multiple network support (part 1)

### PR DESCRIPTION
**Pull request by Jack-Works**
Related #73

### CODE REVIEW

File: src/social-network-provider/facebook.com/ui-provider.ts [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    resolveLastRecognizedIdentity: resolveLastRecognizedIdentityFacebook,
+    injectPostBox: injectPostBoxFacebook,
+    injectWelcomeBanner: injectWelcomeBannerFacebook,
+    injectPostComments: injectPostCommentsDefault(),
+    injectCommentBox: injectCommentBoxDefault(async function onPasteToCommentBoxFacebook(encryptedComment, current) {
+        const root = current.rootNode
+        selectElementContents(root.querySelector('[contenteditable]')!)
+        dispatchCustomEvents('paste', encryptedComment)
+        await sleep(200)
+        if (root.innerText.match(encryptedComment)) 'Okay'
```
Comment (Misaka-0x447f) [inline src/social-network-provider/facebook.com/ui-provider.ts]: expression 'Okay' does nothing

File: src/setup.ts [hunk 2]
```diff
@@ -1,6 +1,11 @@
+import './social-network-provider/facebook.com/ui-provider'
+import './social-network-provider/facebook.com/worker-provider'
 export function backgroundSetup() {
     Object.assign(window, {
         elliptic: require('elliptic'),
     })
 }
-export function uiSetup() {}
+import { activateSocialNetworkUI } from './social-network/ui'
```
Comment (Misaka-0x447f) [inline src/setup.ts]: ts cannot found module.

File: src/network/gun/version.2.ts [hunk 3]
```diff
@@ -21,3 +25,41 @@ export interface ApplicationStateInGunVersion2 {
...
+) {
+    // Store AES key to gun
+    const stored: {
+        [postIdentifier: string]: PublishedAESKey
+    } = {}
+    for (const k of receiversKeys) {
+        stored[k.name] = k.key
+    }
+    console.log('Save to gun', postIdentifier, receiversKeys)
+    // await gun
```
[the remark this answers is not in the document]
Comment (Jack-Works) [reply src/network/gun/version.2.ts]: These codes are key exchange v2, will use it later.

**Comment by SunriseFox** [comment]
+ live-selector seems not correctly selecting the initially-existed posts
+ posts are not decrypted
+ three pages of `m.facebook.com` are opened in background when viewing user profile page
+ extension presists after chrome exits

**Comment by Misaka-0x447f** [comment]
- [ ] tabs auto open in the background infinitly.

File: src/extension/background-script/PeopleService.ts [hunk 4]
```diff
@@ -122,7 +123,7 @@ export async function resolveIdentityAtFacebook(identifier: PersonIdentifier) {
         removeMyIdentityAtDB(self)
```
[the remark this answers is not in the document]
Comment (Jack-Works) [reply src/extension/background-script/PeopleService.ts]: Yes, there is no need to wait for the result and the failure is acceptable

File: src/network/gun/version.2.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (Misaka-0x447f) [reply src/network/gun/version.2.ts]: Emmm I means comments contains unused code

**Pull request merged**